### PR TITLE
updated checkout action version

### PR DESCRIPTION
**Pull request by james-garriss**
## 🗣 Description ##

Updated version of a GitHub action.

## 💭 Motivation and context ##

Updating to current version reduces security threat vectors.

## 🧪 Testing ##

The GitHub Action ran, and the code was checked out.  Success!

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

### CODE REVIEW

File: .github/workflows/run_psscriptanalyzer.yaml
Comment (crutchfield) [inline .github/workflows/run_psscriptanalyzer.yaml:1]: Why v3 instead of v4?

Comment (crutchfield) [reply .github/workflows/run_psscriptanalyzer.yaml:1]: @james-garriss What are your thoughts?

Comment (james-garriss) [reply .github/workflows/run_psscriptanalyzer.yaml:1]: Updated all workflows to use checkout v4 on $669.

Comment (james-garriss) [reply .github/workflows/run_psscriptanalyzer.yaml:1]: Checkout still occurs on all workflows AFAICT.

Comment (james-garriss) [reply .github/workflows/run_psscriptanalyzer.yaml:1]: @crutchfield , I think it's done.

**Review by crutchfield — APPROVED** [review]
Looks good to me.

**Comment by james-garriss** [comment]
@nanda-katikaneni , this PR is done.

**Pull request merged**